woocommerce/woocommerce-ios
# Shipping Labels M2: handle the selected value in the package list

Closes #3964 
Closes #3966
Closes #3965 

## Description
In this PR there are a lot of deletions because I removed a lot of superfluous code after the refactoring, and now the data of the Package Flow are shared between the views as environment property, making the code lighter and more understandable.
I also handled the value selected in the Package List screen and moved the fetching of the existing packages from the remote endpoint in the main view model.


## Changes
- Fixed the `WeightFormatter`, which now is able to handle the case when the weight is an empty string + tests.
- Moved some Shipping Labels classes under a new folder.
- Deleted the `ShippingLabelPackageListViewModel` because it is no more needed.
- Deleted the middleware view `ShippingLabelPackageSelected` (and its view model) that will be reimplemented in M4, since now it is useless.
- Now the `ShippingLabelPackageList` view takes the `ShippingLabelPackageDetailsViewModel` that is declared as `EnvironmentObject`, and that will be updated dynamically in the entire Package Selection flow.
- `ShippingLabelPackageDetails` now handle the selected package, showing the ID of the selected package. The final value will be shown later and will be part of another PR.
- In `ShippingLabelPackageDetails` I moved the `sampleOrder()` method under the view model.
- `ShippingLabelPackageDetailsViewModel`: massive refactoring, adding the fetching of the existing packages from API and the selected package + unit tests.
- `TitleAndValueRow`: fixed the selection, adding the content shape.

## Testing
1. Launch the app in debug mode (because of the feature flag).
2. Open an order detail.
3. Press the button "Create Shipping Label".
4. Press the button "Continue" in Create Shipping Label form, under the **Ship From** cell. Confirm the address.
5. Press the button "Continue" in Create Shipping Label form, under the **Ship To** cell. Confirm the address.
6. Press the button "Continue" in Create Shipping Label form, under the **Package Details** cell.
7. You should see the new screen that shows the Package Details screen and all the items that will be shipped.
8. Press the "Package selected" row. -> You should see the new screen with the list of the existing packages, with Custom Packages on top if any.
9. Select a package, and press "Done". -> You should see the selected package ID under the "Package Selected" row.

## GIF
<img src="https://user-images.githubusercontent.com/495617/115851103-e1446300-a426-11eb-9336-ff96a985b223.gif" width=300 />



Update release notes:
